R/annotation.r

```diff
@@ -46,15 +46,17 @@ annotate <- function(geom, x = NULL, y = NULL, xmin = NULL, xmax = NULL,
 
   # Check that all aesthetic have compatible lengths
   lengths <- vapply(aesthetics, length, integer(1))
-  unequal <- length(unique(setdiff(lengths, 1L))) > 1L
-  if (unequal) {
+  n <- unique(setdiff(lengths, 1L))
+  if (length(n) == 0L) n <- 1L # if all lengths are equal to 1L then above line fails, this fixes that
```

Comment:
Why does it fail?

Reply:
Maybe a poorly formulated comment. The purpose of line 49 is to find the unique lengths that are not `1L`. However, if `lengths == c(1L)`, then we get `n <- numeric(0)` in line 49, and we need `n <- 1L` in this case.